convert2rhel/subscription.py

```diff
@@ -245,9 +279,15 @@ def from_tool_opts(cls, tool_opts):
 
             registration_attributes["password"] = password
 
-        if tool_opts.serverurl:
-            loggerinst.debug("    ... using custom RHSM URL")
-            registration_attributes["server_url"] = tool_opts.serverurl
+        if tool_opts.rhsm_hostname:
+            loggerinst.debug("    ... using custom RHSM hostname")
+            self.rhsm_hostname = tool_opts.rhsm_hostname
+        if tool_opts.rhsm_port:
+            loggerinst.debug("    ... using custom RHSM port")
+            self.rhsm_port = tool_opts.rhsm_port
+        if tool_opts.rhsm_prefix:
+            loggerinst.debug("    ... using custom RHSM prefix")
+            self.rhsm_prefix = tool_opts.rhsm_prefix
```

Comment:
`self` is not defined here, it's a `classmethod`

```suggestion
            cls.rhsm_hostname = tool_opts.rhsm_hostname
        if tool_opts.rhsm_port:
            loggerinst.debug("    ... using custom RHSM port")
            cls.rhsm_port = tool_opts.rhsm_port
        if tool_opts.rhsm_prefix:
            loggerinst.debug("    ... using custom RHSM prefix")
            cls.rhsm_prefix = tool_opts.rhsm_prefix
```

Reply:
Michal actually had a slightly different solution here: https://github.com/oamg/convert2rhel/pull/540#discussion_r943563943

It's a little tricky, if you set `cls.rhsm_hostname`, then you are setting the attribute on the class.  And then when the instance of the class is initialized in `__init__()`, the instance attribute will mask the class attribute.  (ie: this would end up being None because it is set to None by `__init__()`).  Adding the values to `registration_attributes` which are the parameters which are passed into `__init__()` is the right way to fix my mistake.